lucetc/lucetc/main.rs

```diff
@@ -39,20 +41,45 @@ fn main() {
 
     let opts = Options::get().unwrap();
 
-    if let Err(err) = run(&opts) {
-        match opts.error_style {
-            ErrorStyle::Human => {
+    let compile_result = run(&opts);
+
+    let encountered_error = compile_result.is_err();
+
+    let timing_info = if opts.report_times {
+        Some(TimingInfo::collect())
+    } else {
+        None
+    };
+
+    match opts.error_style {
+        ErrorStyle::Human => {
+            if let Err(err) = &compile_result {
                 eprintln!("Error: {}\n", err);
                 if let Some(cause) = err.as_fail().cause() {
                     eprintln!("{}", cause);
                 }
             }
-            ErrorStyle::Json => {
-                let errs: Vec<SerializedLucetcError> = vec![err.into()];
-                let json = serde_json::to_string(&errs).unwrap();
-                eprintln!("{}", json);
+            if let Some(timing_info) = timing_info {
+                // reproduce the header from cranelift-codegen
+                eprintln!("======== ========  ==================================");
+                eprintln!("   Total     Self  Pass");
+                eprintln!("-------- --------- ----------------------------------");
+                eprintln!("{}", timing_info);
             }
         }
+        ErrorStyle::Json => {
+            let errs: Option<Vec<SerializedLucetcError>> =
+                compile_result.err().map(|err| vec![err.into()]);
+            let result = json!({
```

Comment:
this is a breaking change in output format for anyone mechanically processing lucetc output today, but the array of errors was probably a bit too rigid anyway?